R/save_data.R

```diff
@@ -13,51 +15,44 @@
 #'   system.file("examples", "dettl_config.yml", package = "dettl")
 #' )
 #' import <- dettl(file.path(path, "person_information"), "test")
-#' run_import(import, "extract", dry_run = TRUE)
+#' import$extract()
+#' import$transform()
 #' t <- tempfile()
-#' save_extracted_data(import, t)
-save_extracted_data <- function(import, file) {
-  save_data(import, file, "extracted")
-}
-
-#' Save transformed data
-#'
-#' Saves any transformed data as separate sheet of an xlsx file.
-#'
-#' @param import The import object to save the transformed data for.
-#' @param file File path at which to save the data.
-#'
-#' @export
-#'
-#' @examples
-#' path <- dettl:::prepare_test_import(
-#'   system.file("examples", "person_information", package = "dettl"),
-#'   system.file("examples", "dettl_config.yml", package = "dettl")
-#' )
-#' import <- dettl(file.path(path, "person_information"), "test")
-#' run_import(import, c("extract", "transform"), dry_run = TRUE)
-#' t <- tempfile()
-#' save_transformed_data(import, t)
-save_transformed_data <- function(import, file) {
-  save_data(import, file, "transformed")
-}
-
-save_data <- function(import, file, stage) {
+#' dettl_save(import, t, "extract")
+#' t2 <- tempfile()
+#' dettl_save(import, t2, "transform")
+#' t3 <- tempfile()
+#' dettl_save(import, t3, c("extract", "transform"))
+dettl_save <- function(import, file, stage) {
   if (!inherits(import, "DataImport")) {
-    stop(sprintf("Can't save %s data for non-DataImport object.", stage))
+    stop(sprintf("Can't save %s data for non-DataImport object.",
+                 paste(stage, collapse = " and ")))
   }
   dir.create(dirname(file), FALSE, TRUE)
 
-  if (stage == "extracted") {
+  if (all(stage == "extract")) {
     data <- import$get_extracted_data()
-  } else if (stage == "transformed") {
+  } else if (all(stage == "transform")) {
     data <- import$get_transformed_data()
+  } else if ("extract" %in% stage && "transform" %in% stage) {
+    extract <- import$get_extracted_data()
+    ## Data written to xlsx must have unique names
+    names(extract) <- lapply(names(extract),
+                             function(x) paste0("extracted_", x))
+    transform <- import$get_transformed_data()
+    names(transform) <- lapply(names(transform),
+                               function(x) paste0("transformed_", x))
+    data <- c(extract, transform)
   }
 
   if (is.null(data)) {
-    stop(sprintf("Can't save %s data as stage has not been run.", stage))
+    stop(sprintf("Can't save %s data as stage has not been run.",
+         paste(stage, collapse = ", ")))
   }
   writexl::write_xlsx(data, file)
-  message(sprintf("Saved %s data to %s", stage, file))
+  messages <- sprintf("Saved %s data to %s", stage, file)
+  for (message in messages) {
```

Comment:
or
```
message(paste(messages, collapse = "\n"))
```

Reply:
That's nicer as we can avoid the for loop. I've updated that.